app/src/main/java/com/github/palFinderTeam/palfinder/utils/generics/CachedRepository.kt

```diff
@@ -0,0 +1,76 @@
+package com.github.palFinderTeam.palfinder.utils.generics
+
+import android.icu.util.Calendar
+import com.github.palFinderTeam.palfinder.cache.DictionaryCache
+import com.github.palFinderTeam.palfinder.utils.Response
+import com.github.palFinderTeam.palfinder.utils.context.ContextService
+import com.github.palFinderTeam.palfinder.utils.evictAfterXMinutes
+import com.github.palFinderTeam.palfinder.utils.time.TimeService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+class CachedRepository<T : FirebaseObject>(
+    private val name: String,
+    private val clazz: Class<T>,
+    private val rp: Repository<T>,
+    private val time: TimeService,
+    private val contextProvider: ContextService
+) : Repository<T> {
+    private var cache =
```

Comment:
Same comment for the over cached services, is it important to have var here ? (I don't have androidstudio open so I cannot check if there is a warning).